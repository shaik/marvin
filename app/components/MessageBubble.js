import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

export default function MessageBubble({ role, text, rtl = false, timestamp }) {
  const isUser = role === 'user';
  const containerAlign = isUser ? 'flex-end' : 'flex-start';
  const bubbleStyle = isUser ? styles.userBubble : styles.assistantBubble;
  const resolvedDirection = rtl ? 'rtl' : 'ltr';

  return (
    <View style={[styles.row, { justifyContent: containerAlign }]}>
      <View style={[styles.bubble, bubbleStyle, styles.shadow]}>
        <Text
          testID={`bubble-text-${rtl ? 'rtl' : 'ltr'}`}
          style={[styles.text, { writingDirection: resolvedDirection, textAlign: rtl ? 'right' : 'left' }]}
        >
          {text}
        </Text>
        {timestamp ? (
          <Text style={[styles.time, { textAlign: isUser ? 'right' : 'left' }]}>{timestamp}</Text>
        ) : null}
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  row: {
    width: '100%',
    marginVertical: 6,
  },
  bubble: {
    maxWidth: '85%',
    paddingVertical: 8,
    paddingHorizontal: 12,
    borderRadius: 14,
  },
  userBubble: {
    backgroundColor: '#DCFCE7',
    borderTopRightRadius: 4,
  },
  assistantBubble: {
    backgroundColor: '#FFFFFF',
    borderTopLeftRadius: 4,
    borderWidth: 1,
    borderColor: '#E5E7EB',
  },
  text: {
    fontSize: 16,
    color: '#111827',
  },
  rtl: {
    writingDirection: 'rtl',
    textAlign: 'right',
  },
  time: {
    marginTop: 4,
    fontSize: 10,
    color: '#6B7280',
  },
  shadow: {
    shadowColor: '#000',
    shadowOpacity: 0.05,
    shadowRadius: 3,
    shadowOffset: { width: 0, height: 1 },
    elevation: 2,
  },
});


