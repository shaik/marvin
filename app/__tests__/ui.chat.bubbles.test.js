import React from 'react';
import { render } from '@testing-library/react-native';
import MessageBubble from '../components/MessageBubble';

describe('MessageBubble styling and RTL', () => {
  it('renders user and assistant bubbles (snapshot)', () => {
    const { toJSON } = render(
      <>
        <MessageBubble role="user" text="Hello" />
        <MessageBubble role="assistant" text="Hi there" />
      </>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders RTL via testID when rtl=true', () => {
    const { getByTestId } = render(<MessageBubble role="assistant" text={"שלום"} rtl={true} />);
    expect(getByTestId('bubble-text-rtl')).toBeTruthy();
  });
});


