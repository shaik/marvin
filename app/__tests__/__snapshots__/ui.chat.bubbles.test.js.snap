// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBubble styling and RTL renders user and assistant bubbles (snapshot) 1`] = `
[
  <View
    style={
      [
        {
          "marginVertical": 6,
          "width": "100%",
        },
        {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderRadius": 14,
            "maxWidth": "85%",
            "paddingHorizontal": 12,
            "paddingVertical": 8,
          },
          {
            "backgroundColor": "#DCFCE7",
            "borderTopRightRadius": 4,
          },
          {
            "elevation": 2,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.05,
            "shadowRadius": 3,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#111827",
              "fontSize": 16,
            },
            {
              "textAlign": "left",
              "writingDirection": "ltr",
            },
          ]
        }
        testID="bubble-text-ltr"
      >
        Hello
      </Text>
    </View>
  </View>,
  <View
    style={
      [
        {
          "marginVertical": 6,
          "width": "100%",
        },
        {
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderRadius": 14,
            "maxWidth": "85%",
            "paddingHorizontal": 12,
            "paddingVertical": 8,
          },
          {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#E5E7EB",
            "borderTopLeftRadius": 4,
            "borderWidth": 1,
          },
          {
            "elevation": 2,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.05,
            "shadowRadius": 3,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#111827",
              "fontSize": 16,
            },
            {
              "textAlign": "left",
              "writingDirection": "ltr",
            },
          ]
        }
        testID="bubble-text-ltr"
      >
        Hi there
      </Text>
    </View>
  </View>,
]
`;
